services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/scenario/ratelimit/ProdPpacIosRateLimitStrategy.java

```diff
@@ -22,14 +23,25 @@
   private static final Logger logger = LoggerFactory.getLogger(ProdPpacIosRateLimitStrategy.class);
   static final int VALIDITY_IN_HOURS = 23;
 
+  private final Clock clock;
+
+  @Autowired
```

Comment:
Any particular reason for the use of Autowire here? I always thought one can use Autowire to perform Field Injection instead of constructor injection. My opinion is that the usage of Autowire should not be placed here.

Reply:
Marks a constructor [...] as to be autowired by Spring's dependency injection facilities: https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/factory/annotation/Autowired.html
This is also according to some code examples I found on the net...

Reply:
Ok I understand. then other change I suggested is not necessary

Reply:
How does this then work from inside the tests?

Reply:
Yes. So this is the default for everything. Right now, we only need the "special" constructor to test a specific edge-case. So it only is used for that single test.